dbt/adapters/duckdb/environments/local.py

```diff
@@ -58,6 +58,9 @@ def notify_closed(self):
             if self.handle_count == 0 and not self._keep_open:
                 self.close()
 
+    def cancel(cls, connection: Connection):
+        connection.handle.cursor().interrupt()
```

Comment:
Is this a new feature of MD or of DuckDB in general?

Reply:
This is a duckdb feature: https://github.com/duckdb/duckdb/pull/7895
I also manually tested it with MD and saw the expected behavior.